app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_catcher/stream.rb

```diff
@@ -17,32 +15,27 @@ def stop
     @stop_polling = true
   end
 
-  def fake_events
-    [
-      OpenStruct.new(:name       => %w[instance_power_on instance_power_off].sample,
-                     :id         => Time.zone.now.to_i,
-                     :timestamp  => Time.zone.now,
-                     :vm_ems_ref => [1, 2].sample),
-      OpenStruct.new(:name       => %w[instance_power_on instance_power_off].sample,
-                     :id         => Time.zone.now.to_i + 1,
-                     :timestamp  => Time.zone.now,
-                     :vm_ems_ref => [1, 2].sample),
-    ]
-  end
-
   def poll
-    @ems.with_provider_connection do |_connection|
-      catch(:stop_polling) do
-        loop do
-          fake_events.each do |activity|
-            throw :stop_polling if @stop_polling
-            yield activity.to_h
-          end
-          sleep @poll_sleep
-        end
-      rescue => exception
-        raise ProviderUnreachable, exception.message
+    from_time = Time.now.utc.to_i
+
+    loop do
+      ems.with_provider_connection(:service => "PCloudEventsApi") do |api|
+        # Could also use IBM Cloud Activity Tracker like the VPC provider
+        events = api.pcloud_events_getsince(cloud_instance_id, from_time).events
```

Comment:
Link to the PowerVS Events API:
https://cloud.ibm.com/apidocs/power-cloud#pcloud-events-getsince

Unfortunately, in my testing I'm always getting an empty `metadata` object, which is supposed to include the VM instance ID. I'm following up with the API devs. If this doesn't work the options would be to parse the VM instance name from the `message` string, or use the IBM Cloud Activity Tracker API like VPC does.

Reply:
Decided to go with parsing out the `message` strings.